cardano-api/src/Cardano/Api/IPC/Monad.hs

```diff
@@ -52,44 +59,56 @@ executeLocalStateQueryExpr
   -> Maybe ChainPoint
   -> LocalStateQueryExpr (BlockInMode mode) ChainPoint (QueryInMode mode) () IO a
   -> IO (Either AcquiringFailure a)
-executeLocalStateQueryExpr connectInfo mpoint f = do
-  tmvResultLocalState <- newEmptyTMVarIO
+executeLocalStateQueryExpr connectInfo mpoint f =
+  runOopsInEither $ executeLocalStateQueryExpr_ connectInfo mpoint (lift f)
+
+-- | Execute a local state query expression.
+executeLocalStateQueryExpr_
+  :: forall e mode a . ()
+  => e `CouldBe` AcquiringFailure
+  => LocalNodeConnectInfo mode
+  -> Maybe ChainPoint
+  -> ExceptT (Variant e) (LocalStateQueryExpr (BlockInMode mode) ChainPoint (QueryInMode mode) () IO) a
+  -> ExceptT (Variant e) IO a
+executeLocalStateQueryExpr_ connectInfo mpoint f = do
+  tmvResultLocalState <- liftIO (newEmptyTMVarIO @(Either (Variant e) a))
   let waitResult = readTMVar tmvResultLocalState
 
-  connectToLocalNodeWithVersion
+  liftIO $ connectToLocalNodeWithVersion
     connectInfo
     (\ntcVersion ->
       LocalNodeClientProtocols
       { localChainSyncClient    = NoLocalChainSyncClient
-      , localStateQueryClient   = Just $ setupLocalStateQueryExpr waitResult mpoint tmvResultLocalState ntcVersion f
+      , localStateQueryClient   = Just $ setupLocalStateQueryExpr_ waitResult mpoint tmvResultLocalState ntcVersion f
       , localTxSubmissionClient = Nothing
       , localTxMonitoringClient = Nothing
       }
     )
 
-  first toAcquiringFailure <$> atomically waitResult
+  ExceptT . return =<< liftIO (atomically waitResult)
```

Comment:
Again an important feature of `oops` is it does not require us to restructure our code.  The `oops` version is structurally very similar to the non-`oops` one.